### ISSUE TITLE
Interactable code

### ISSUE THREAD

**Issue by Wadarass** [RyannR9375/Dialouge-System-2023#1]
The dialoguemanager is a singleton.  No need for findObjectsOfType. 

Also for the dialogue line to start.  Maybe have the item specify the name of the line? Instead of a dialogue database.  The item shouldn’t care about the database, only the dialogue manager.

**Comment by RyannR9375** [comment]
Hey Raymond,

Changed the findObjectOfType to DialogueManager.Instance
is this okay/best approach?

Also I was thinking the NPC can take in a Dialogue Database so that we
could have different Dialogue trees for different NPC's?
Or should I have the Dialogue Manager take in multiple databases for
different conversations, and then have the NPC's reference that database
from the manager?

**Comment by Wadarass** [comment]
> Changed the findObjectOfType to DialogueManager.Instance
> is this okay/best approach?

yes that is what I would expect.  That's the reason it's a singleton don't need to do findObjectOfType.

> Also I was thinking the NPC can take in a Dialogue Database so that we
> could have different Dialogue trees for different NPC's?
> Or should I have the Dialogue Manager take in multiple databases for
> different conversations, and then have the NPC's reference that database
> from the manager?
This is a good instinct but you probably want something else to manage the loading and unloading of dialogue databases for different NPCs, or levels.  Main thing is you need to initialize the dialogue Manager with the new dialogue database so that probably should be separate from the actual items that trigger dialogue.  They should just use the name of the dialogue they want to start and then pass that to the manager.
